src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -17,15 +17,30 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public abstract class TextLoader
     {
+        /// <summary>
+        /// Immutable workspace passed to public APIs of this type that take <see cref="Workspace"/> instance.
+        /// We no longer pass the actual workspace around. To avoid breaking dervied types that may access the workspace entirely
+        /// we pass in a dummy <see cref="ImmutableEmptyWorkspace"/>. This workspace won't have the services that would have been previously available
+        /// if used e.g. in VS layer though so a break is still possible.
+        /// </summary>
+        private protected static readonly Workspace DummyWorkspace = new ImmutableEmptyWorkspace(Host.Mef.MefHostServices.DefaultHost, WorkspaceKind.Custom);
```

Comment:
The use of DefaultHost here means we end up spinning up an extra MEF container which usually breaks RPS. If the goal of this is to not have services, then we could make a placeholder host that just exposes nothing at all. This bit is a must-fix.

Reply:
I see. Makes sense.